### PR TITLE
feat: Add chaining RxJava

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Add chaining RxJava on loading order details, signing up, editting user profile
- Remove unused methods

Fixes: #2116

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt [hunk 1]
```diff
@@ -37,43 +38,18 @@ class EditProfileViewModel(
...
-            ), id
-        )
+        val updateProfileObservable: Single<User> =
+            if (encodedImage.isNullOrEmpty())
+                authService.updateUser(User(id = id, firstName = firstName, lastName = lastName, avatarUrl = null,
+                    details = details), id)
+            else
+                authService.uploadImage(UploadImage(encodedImage)).flatMap {
+                    authService.updateUser(User(id = id, firstName = firstName, lastName = lastName,
+                        avatarUrl = it.url, details = details), id)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: Repetition, please use data class copy constructor

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt [hunk 2]
```diff
@@ -36,14 +37,21 @@ class SignUpViewModel(
     fun signUp(signUp: SignUp) {
         if (!isConnected()) return
 
-        compositeDisposable += authService.signUp(signUp)
+        val signUpObservable: Single<User> = authService.signUp(signUp).flatMap {
+            email = signUp.email
+            password = signUp.password
+            authService.login(email.nullToEmpty(), password.nullToEmpty()).flatMap {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt]: Chaining should not be nested

File: app/src/main/java/org/fossasia/openevent/general/order/OrderDetailsViewModel.kt [hunk 3]
```diff
@@ -54,31 +55,20 @@ class OrderDetailsViewModel(
     fun loadAttendeeDetails(orderId: Long) {
         if (orderId == -1L) return
 
-        compositeDisposable += orderService.getOrderById(orderId)
+        val attendeeObservable: Single<List<Attendee>> = orderService.getOrderById(orderId).flatMap { order ->
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrderDetailsViewModel.kt]: Separate operators to each on new line

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt [hunk 4]
```diff
@@ -36,14 +37,22 @@ class SignUpViewModel(
...
         if (!isConnected()) return
 
-        compositeDisposable += authService.signUp(signUp)
+        val signUpAndLogInObservable: Single<LoginResponse> = authService.signUp(signUp).flatMap {
+            email = signUp.email
+            password = signUp.password
+            authService.login(email.nullToEmpty(), password.nullToEmpty())
+        }
+
+        val profileAfterLogInObservable: Single<User> = signUpAndLogInObservable.flatMap { authService.getProfile() }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt]: No need to introduce a new variable, chain the flatMap calls

File: app/src/main/java/org/fossasia/openevent/general/order/OrderDetailsViewModel.kt [hunk 5]
```diff
@@ -54,31 +55,22 @@ class OrderDetailsViewModel(
     fun loadAttendeeDetails(orderId: Long) {
         if (orderId == -1L) return
 
-        compositeDisposable += orderService.getOrderById(orderId)
+        val attendeesObservable: Single<List<Attendee>> = orderService
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrderDetailsViewModel.kt]: No need for a variable

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt [hunk 6]
```diff
@@ -37,43 +38,17 @@ class EditProfileViewModel(
...
+        val user = User(id = id, firstName = firstName, lastName = lastName, avatarUrl = null, details = details)
+        val updateProfileObservable: Single<User> =
+            if (encodedImage.isNullOrEmpty())
+                authService.updateUser(user, id)
+            else
+                authService.uploadImage(UploadImage(encodedImage)).flatMap {
+                    authService.updateUser(user.copy(avatarUrl = it.url), id)
+                }
+
+        compositeDisposable += updateProfileObservable
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: No need of updateProfileObservable, please remove any obsolete variable

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/SignUpViewModel.kt]: Still not resolved. Chaining should not be nested. That's the point of chaining

**Comment by anhanh11001** [comment]
@iamareebjamal, please review

**Pull request merged**